sql/src/main/java/io/crate/operation/tablefunctions/GenerateSeriesFunction.java

```diff
@@ -0,0 +1,279 @@
+
+        type = DataTypes.FLOAT;
+
+        implTwoArgs.put(type, new FloatGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type)), type, FunctionInfo.Type.TABLE)));
+        implThreeArgs.put(type, new FloatGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type, type)), type, FunctionInfo.Type.TABLE)));
+
+        type = DataTypes.DOUBLE;
+
+        implTwoArgs.put(type, new DoubleGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type)), type, FunctionInfo.Type.TABLE)));
+        implThreeArgs.put(type, new DoubleGenerateSeriesFunctionImplementation(new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(type, type, type)), type, FunctionInfo.Type.TABLE)));
+    }
+
+
+    public static void register(TableFunctionModule module){
+        module.register(NAME, new BaseFunctionResolver(
+            Signature.numArgs(2,3).and(Signature.withLenientVarArgs(Signature.ArgMatcher.NUMERIC))) {
+
+            @Override
+            public FunctionImplementation getForTypes(List<DataType> dataTypes) throws IllegalArgumentException {
+                Map<DataType, FunctionImplementation> impl = dataTypes.size() == 2 ? implTwoArgs : implThreeArgs;
+                DataType first = dataTypes.get(0);
+                DataType third = dataTypes.size() > 2 ? dataTypes.get(2) : DataTypes.INTEGER;
+
+                DataType implType;
+                if(NUMERIC_WITH_DECIMAL.contains(first) || NUMERIC_WITH_DECIMAL.contains(third))
+                    implType = dataTypes.contains(DataTypes.DOUBLE) || dataTypes.contains(DataTypes.LONG) ? DataTypes.DOUBLE : DataTypes.FLOAT;
+                else
+                    implType = dataTypes.contains(DataTypes.LONG) ? DataTypes.LONG : DataTypes.INTEGER;
+                return impl.get(implType);
+            }
+        });
+    }
+
+    static abstract class GenerateSeriesFunctionImplementation<T extends Number> implements TableFunctionImplementation {
+
+        private final FunctionInfo info;
+        private final Map<ColumnIdent, Reference> columnMap;
+        private final Collection<Reference> columns;
+
+        protected abstract T plus(T v1, T v2);
+        protected abstract T minus(T v1, T v2);
+        protected abstract T divide(T v1, T v2);
+
+        public GenerateSeriesFunctionImplementation(FunctionInfo info) {
+            this.info = info;
+
+            ColumnIdent columnIdent = new ColumnIdent(NAME);
+            Reference reference = new Reference(new ReferenceIdent(TABLE_IDENT, columnIdent), RowGranularity.DOC, info.returnType());
+            columnMap = Collections.singletonMap(columnIdent, reference);
+            columns = Collections.singletonList(reference);
+        }
+
+        @Override
+        public FunctionInfo info() {
+            return info;
+        }
+
```

Comment:
you need to overriede normalizeSymbol so that the actual arguments match the argument types of the function symbol used to represent the implementation